Index: ocdmRialto/CMakeLists.txt
===================================================================
--- ocdmRialto.orig/CMakeLists.txt
+++ ocdmRialto/CMakeLists.txt
@@ -14,7 +14,6 @@ set( CMAKE_INCLUDE_CURRENT_DIR ON )
 
 # Set local path for searching for cmake files
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
-find_package(WPEFramework)
 include_directories(include)
 
 set(LIB_OCDM_RIALTO_SOURCES
Index: ocdmRialto/source/OpenCDMSession.cpp
===================================================================
--- ocdmRialto.orig/source/OpenCDMSession.cpp
+++ ocdmRialto/source/OpenCDMSession.cpp
@@ -1,5 +1,7 @@
 #include <OpenCDMSession.h>
-#include <WPEFramework/core/Trace.h>
+//#include <WPEFramework/core/Trace.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 
 OpenCDMSession::OpenCDMSession(std::shared_ptr<rialto::client::ContentDecryptionBackend> cdm, const std::string& keySystem,
     const LicenseType& sessionType,
@@ -158,4 +160,4 @@ rialto::client::ContentDecryptionBackend
     }
 
     return initDataType;
-}
\ No newline at end of file
+}
Index: ocdmRialto/source/open_cdm.cpp
===================================================================
--- ocdmRialto.orig/source/open_cdm.cpp
+++ ocdmRialto/source/open_cdm.cpp
@@ -1,9 +1,12 @@
+#include <cassert>
 #include <opencdm/open_cdm.h>
 
 
 #include <OpenCDMSession.h>
 #include <OpenCDMSystem.h>
-#include <WPEFramework/core/Trace.h>
+//#include <WPEFramework/core/Trace.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 
 
 extern "C" void* createContentDecryptionBackend(void* client);
Index: ocdmRialto/source/open_cdm_adapter.cpp
===================================================================
--- ocdmRialto.orig/source/open_cdm_adapter.cpp
+++ ocdmRialto/source/open_cdm_adapter.cpp
@@ -1,21 +1,7 @@
 #include <opencdm/open_cdm_adapter.h>
 
-OpenCDMError opencdm_gstreamer_session_decrypt_ex(struct OpenCDMSession* session, GstBuffer* buffer, GstBuffer* subSample, const uint32_t subSampleCount,
-                                                   GstBuffer* IV, GstBuffer* keyID, uint32_t initWithLast15, GstCaps* caps)
-{
-    //Decrypting is done on Rialto server side after pulling GstBuffers from Rialto sinks
-    //Do nothing now
-    return ERROR_NONE;
-}
-
-
 OpenCDMError opencdm_gstreamer_session_decrypt(struct OpenCDMSession* session, GstBuffer* buffer, GstBuffer* subSample, const uint32_t subSampleCount,
     GstBuffer* IV, GstBuffer* keyID, uint32_t initWithLast15)
 {
-    return opencdm_gstreamer_session_decrypt_ex(session, buffer, subSample, subSampleCount, IV, keyID, initWithLast15, nullptr);
-}
-
-OpenCDMError opencdm_gstreamer_transform_caps(GstCaps** caps)
-{
     return ERROR_NONE;
-}
\ No newline at end of file
+} 
