Index: ocdmRialto/CMakeLists.txt
===================================================================
--- ocdmRialto.orig/CMakeLists.txt
+++ ocdmRialto/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.10.3)
-project(ocdmRialto)
+project(ocdmRialto VERSION 0.1.0)
+set(ocdmRialto_SOVERSION 0)
 
 set(CMAKE_CXX_STANDARD 14)
 enable_language(CXX)
@@ -24,19 +25,26 @@ set(LIB_OCDM_RIALTO_SOURCES
         source/OpenCDMSession.cpp)
 
 set(LIB_OCDM_RIALTO_PUBLIC_HEADERS
+        include/opencdm/open_cdm.h
+        include/opencdm/open_cdm_adapter.h
         )
 
 add_library(ocdmRialto SHARED ${LIB_OCDM_RIALTO_SOURCES} )
 
+set_target_properties(ocdmRialto PROPERTIES VERSION ${PROJECT_VERSION})
+
 install(TARGETS ocdmRialto LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 target_link_libraries(ocdmRialto PUBLIC
        RialtoClient)
 
+configure_file(
+       "pkg-config/ocdmRialto.pc.in"
+       "pkg-config/ocdmRialto.pc"
+       @ONLY)
 
-InstallPackageConfig(
-        TARGETS ocdmRialto
-        DESCRIPTION "OCDM library using Rialto")
-
+install(FILES ${LIB_OCDM_RIALTO_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/opencdm)
 
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pkg-config/ocdmRialto.pc"
+       DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
Index: ocdmRialto/pkg-config/ocdmRialto.pc.in
===================================================================
--- /dev/null
+++ ocdmRialto/pkg-config/ocdmRialto.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name:  ocdmRialto
+Description: OCDM implementation based on librialtoclient library
+Version: @JSONCPP_VERSION@
+Libs: -L${libdir} -locdmRialto
+Cflags: -I${includedir}
+
