Source: Liberty Global B.V.
Index: v0.1.0+gitAUTOINC+4015e49763-r0/git/sloci-image
===================================================================
--- v0.1.0+gitAUTOINC+4015e49763-r0.orig/git/sloci-image
+++ v0.1.0+gitAUTOINC+4015e49763-r0/git/sloci-image
@@ -269,13 +269,13 @@ oci_image_config() {
 	  "architecture": "$(oci_arch $CFG_ARCH)",
 	  "os": "$CFG_OS",
 	  "config": {
-	    ${CFG_USER:+"\"User\": $(json_string "$CFG_USER"),"}
+	    ${CFG_USER:+"$(json_string "User"): $(json_string "$CFG_USER"),"}
 	    "ExposedPorts": $(json_pseudoarray "$CFG_PORTS"),
 	    "Env": $(json_string_array "$CFG_ENV"),
 	    "Entrypoint": $(json_string_array "$CFG_ENTRYPOINT"),
 	    "Cmd": $(json_string_array "$CFG_CMD"),
 	    "Volumes": $(json_pseudoarray "$CFG_VOLUMES"),
-	    ${CFG_WORKING_DIR:+"\"WorkingDir\": $(json_string "$CFG_WORKING_DIR"),"}
+	    ${CFG_WORKING_DIR:+"$(json_string "WorkingDir"): $(json_string "$CFG_WORKING_DIR"),"}
 	    "Labels": $(json_string_map "$CFG_LABELS")
 	  },
 	  "rootfs": {
@@ -425,6 +425,6 @@ if [ "$OUT_TYPE" = tar ]; then
 	file_name="$IMAGE_NAME-$CFG_REF_NAME-$CFG_ARCH"
 	file_name="$file_name${CFG_ARCH_VARIANT:+"-$CFG_ARCH_VARIANT"}-$CFG_OS.oci-image"
 
-	tar ${DEBUG:+-v} -cf "$file_name.tar" "$IMAGE_NAME"
+	cd $IMAGE_NAME && tar ${DEBUG:+-v} -cf "../$file_name.tar" "."
 	rm -Rf "$IMAGE_NAME"
 fi
