Source: Liberty Global Service B.V.
Index: git/src/third_party/starboard/rdk/shared/player/player_create.cc
===================================================================
--- git.orig/src/third_party/starboard/rdk/shared/player/player_create.cc
+++ git/src/third_party/starboard/rdk/shared/player/player_create.cc
@@ -61,20 +61,6 @@ SbPlayerCreate(SbWindow window,
     return kSbPlayerInvalid;
   }
 
-  if (audio_codec != kSbMediaAudioCodecNone &&
-      !third_party::starboard::rdk::shared::media::
-      GstRegistryHasElementForMediaType(audio_codec)) {
-    SB_LOG(ERROR) << "Unsupported audio codec " << audio_codec;
-    return kSbPlayerInvalid;
-  }
-
-  if (video_codec != kSbMediaVideoCodecNone &&
-      !third_party::starboard::rdk::shared::media::
-      GstRegistryHasElementForMediaType(video_codec)) {
-    SB_LOG(ERROR) << "Unsupported video codec " << video_codec;
-    return kSbPlayerInvalid;
-  }
-
   if (output_mode != kSbPlayerOutputModePunchOut) {
     SB_LOG(ERROR) << "Unsupported player output mode " << output_mode;
     return kSbPlayerInvalid;
