Source: Liberty Global Service B.V.
Index: starboard/src/third_party/starboard/rdk/shared/ess_input.cc
===================================================================
--- starboard.orig/src/third_party/starboard/rdk/shared/ess_input.cc
+++ starboard/src/third_party/starboard/rdk/shared/ess_input.cc
@@ -63,7 +63,7 @@ constexpr SbTime kKeyRepeatTime = 50 * k
 SbKey KeyCodeToSbKey(uint16_t code) {
   switch (code) {
     case KEY_BACKSPACE:
-      return kSbKeyBack;
+      return kSbKeyEscape;
     case KEY_DELETE:
       return kSbKeyDelete;
     case KEY_TAB:
