Index: rialtoClient/CMakeLists.txt
===================================================================
--- rialtoClient.orig/CMakeLists.txt
+++ rialtoClient/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.10.3)
-project(RialtoClient)
+project(RialtoClient VERSION 0.1.0)
+set(RialtoClient_SOVERSION 0)
 
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_CXX_STANDARD 14)
@@ -101,6 +102,7 @@ set(LIB_RIALTO_CLIENT_PUBLIC_HEADERS
 
 add_library(RialtoClient SHARED ${LIB_RIALTO_CLIENT_SOURCES} )
 
+set_target_properties(RialtoClient PROPERTIES VERSION ${PROJECT_VERSION})
 
 target_include_directories(
     RialtoClient
