From: Stefan Verkoyen <sverkoyen.contractor@libertyglobal.com>
Subject: RDKCMF-8834 support latest rialto
Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -35,15 +35,26 @@ option( ENABLE_RIALTO_CLEARKEY "Enabled
 
 # Add client server and ocdm targets
 add_subdirectory( rialtoCommon )
-if( NOT CMAKE_BUILD_TYPE STREQUAL "UnitTests" )
+add_subdirectory( rialtoLogging )
+
+option( RIALTOSERVER "Include rialtoServer" ON )
+option( RIALTOSERVERMANAGER "Include rialtoServerManager" ON )
+option( RIALTOCLIENT "Include rialtoClient" ON )
+option( RIALTOOCDM   "Include ocdmRialto" ON )
+
+if( RIALTOSERVER )
     add_subdirectory( rialtoServer )
+endif()
+if( RIALTOSERVERMANAGER )
+    add_subdirectory( rialtoServerManager )
+endif()
+if( RIALTOCLIENT )
+    add_subdirectory( rialtoGstSinks )
     add_subdirectory( rialtoClient )
+endif()
+if( RIALTOOCDM )
     add_subdirectory( ocdmRialto )
-    add_subdirectory( rialtoGstSinks )
 endif()
-add_subdirectory( rialtoLogging )
-add_subdirectory( rialtoServerManager )
-
 
 # Config and target for building the unit tests
 if( CMAKE_BUILD_TYPE STREQUAL "UnitTests" )
