Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -30,7 +30,6 @@ include( CheckCXXCompilerFlag )
 
 find_package( Rialto REQUIRED )
 find_package( ocdm REQUIRED )
-find_package( WPEFramework REQUIRED )
 find_package( PkgConfig REQUIRED )
 pkg_check_modules( GStreamerApp REQUIRED IMPORTED_TARGET gstreamer-app-1.0 )
 
Index: git/source/OpenCDMSession.cpp
===================================================================
--- git.orig/source/OpenCDMSession.cpp
+++ git/source/OpenCDMSession.cpp
@@ -20,7 +20,8 @@
 #include "OpenCDMSession.h"
 #include "MediaKeysClient.h"
 #include "RialtoGStreamerEMEProtectionMetadata.h"
-#include <WPEFramework/core/Trace.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 #include <gst/base/base.h>
 #include <gst/gst.h>
 #include <gst/gstprotection.h>
Index: git/source/open_cdm.cpp
===================================================================
--- git.orig/source/open_cdm.cpp
+++ git/source/open_cdm.cpp
@@ -23,7 +23,9 @@
 #include <OpenCDMSession.h>
 #include <OpenCDMSystem.h>
 #include <Utils.h>
-#include <WPEFramework/core/Trace.h>
+#include <assert.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 
 OpenCDMSystem *opencdm_create_system(const char keySystem[])
 {
Index: git/source/open_cdm_adapter.cpp
===================================================================
--- git.orig/source/open_cdm_adapter.cpp
+++ git/source/open_cdm_adapter.cpp
@@ -18,7 +18,8 @@
  */
 
 #include <OpenCDMSession.h>
-#include <WPEFramework/core/Trace.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 #include <opencdm/open_cdm_adapter.h>
 
 struct _GstCaps;
Index: git/source/open_cdm_ext.cpp
===================================================================
--- git.orig/source/open_cdm_ext.cpp
+++ git/source/open_cdm_ext.cpp
@@ -20,7 +20,8 @@
 #include <OpenCDMSession.h>
 #include <OpenCDMSystem.h>
 #include <Utils.h>
-#include <WPEFramework/core/Trace.h>
+#define TRACE_L1 printf
+#define TRACE_L2 printf
 #include <opencdm/open_cdm_ext.h>
 
 OpenCDMError opencdm_system_ext_get_ldl_session_limit(struct OpenCDMSystem *system, uint32_t *ldlLimit)
