From: Stefan Verkoyen <sverkoyen.contractor@libertyglobal.com>
Subject: RDKCMF-8813 Add rialto support to reference image
Index: git/CMakeLists.txt
===================================================================
--- git.orig/CMakeLists.txt
+++ git/CMakeLists.txt
@@ -1,6 +1,7 @@
 cmake_minimum_required( VERSION 3.10 )
 project( Rialto LANGUAGES C CXX VERSION 1.0 )
 
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror=return-type")
 
 # Add our local cmake directory to search for components
 set( CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake )
Index: git/rialtoServer/player/Player.cpp
===================================================================
--- git.orig/rialtoServer/player/Player.cpp
+++ git/rialtoServer/player/Player.cpp
@@ -965,6 +965,7 @@ bool Player::decrypt(GstBuffer *buffer,
         return opencdm_gstreamer_session_decrypt(g_ocdmSession, buffer, subsamples, subsamples_count, iv, key,
                                                  initWithLast15) == ERROR_NONE;
     }
+    return false;
 }
 
 #if defined(ENABLE_WESTEROS_SINK_GEOMETRY)
Index: git/rialtoClient/source/ContentDecryptionBackendImpl.cpp
===================================================================
--- git.orig/rialtoClient/source/ContentDecryptionBackendImpl.cpp
+++ git/rialtoClient/source/ContentDecryptionBackendImpl.cpp
@@ -40,6 +40,7 @@ ContentDecryptionBackend* ContentDecrypt
     catch (ContentDecryptionBackendException& e)
     {
         RIALTO_CLIENT_LOG_ERROR("Backend creation failed: %s", e.what());
+        return NULL;
     }
 }
 
