Index: git/rialtoClient/CMakeLists.txt
===================================================================
--- git.orig/rialtoClient/CMakeLists.txt
+++ git/rialtoClient/CMakeLists.txt
@@ -1,3 +1,5 @@
+project(RialtoClient VERSION 0.1.0)
+set(RialtoClient_SOVERSION 0)
 
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_CXX_STANDARD 14)
@@ -89,7 +91,7 @@ target_include_directories(
 set_target_properties(
         RialtoClient
         PROPERTIES LINK_FLAGS "-Wl,--unresolved-symbols=report-all"
-
+                   VERSION ${PROJECT_VERSION}
         )
 
 target_link_libraries(
Index: git/ocdmRialto/CMakeLists.txt
===================================================================
--- git.orig/ocdmRialto/CMakeLists.txt
+++ git/ocdmRialto/CMakeLists.txt
@@ -1,3 +1,5 @@
+project(ocdmRialto VERSION 0.1.0)
+set(ocdmRialto_SOVERSION 0)
 
 set( CMAKE_CXX_STANDARD 14 )
 
@@ -21,6 +23,7 @@ set(LIB_OCDM_RIALTO_PUBLIC_HEADERS
         )
 
 add_library(ocdmRialto SHARED ${LIB_OCDM_RIALTO_SOURCES} )
+set_target_properties(ocdmRialto PROPERTIES VERSION ${PROJECT_VERSION} )
 
 target_include_directories(
         ocdmRialto
