Index: git/ocdmRialto/CMakeLists.txt
===================================================================
--- git.orig/ocdmRialto/CMakeLists.txt
+++ git/ocdmRialto/CMakeLists.txt
@@ -6,7 +6,6 @@ set( CMAKE_CXX_STANDARD 14 )
 set( CMAKE_CXX_STANDARD_REQUIRED ON )
 include( CheckCXXCompilerFlag )
 
-find_package( ocdm REQUIRED )
 find_package( PkgConfig REQUIRED )
 pkg_check_modules( GStreamerApp REQUIRED IMPORTED_TARGET gstreamer-app-1.0 )
 
@@ -50,12 +49,12 @@ include( GNUInstallDirs )
 install(TARGETS ocdmRialto LIBRARY
         DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
-# TODO: replace the .pc generator from WPEFramework with our own locally
-# generated ocdmRialto.pc file
-InstallPackageConfig(
-        TARGETS ocdmRialto
-        DESCRIPTION "OCDM library using Rialto"
-        NO_COMPILE_OPTIONS
-        )
+configure_file(
+       "pkg-config/ocdmRialto.pc.in"
+       "pkg-config/ocdmRialto.pc"
+       @ONLY)
 
+install(FILES ${LIB_OCDM_RIALTO_PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/opencdm)
 
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/pkg-config/ocdmRialto.pc"
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
Index: git/ocdmRialto/pkg-config/ocdmRialto.pc.in
===================================================================
--- /dev/null
+++ git/ocdmRialto/pkg-config/ocdmRialto.pc.in
@@ -0,0 +1,11 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+
+Name:  ocdmRialto
+Description: OCDM implementation based on librialtoclient library
+Version: @JSONCPP_VERSION@
+Libs: -L${libdir} -locdmRialto
+Cflags: -I${includedir}
+
