Index: git/rialtoGstSinks/source/RialtoGSteamerPlugin.cpp
===================================================================
--- git.orig/rialtoGstSinks/source/RialtoGSteamerPlugin.cpp
+++ git/rialtoGstSinks/source/RialtoGSteamerPlugin.cpp
@@ -27,5 +27,5 @@ static gboolean rialto_mse_sinks_init(Gs
            gst_element_register(plugin, "rialtomseaudiosink", sinkRank, RIALTO_TYPE_MSE_AUDIO_SINK);
 }
 
-GST_PLUGIN_DEFINE(GST_VERSION_MAJOR, GST_VERSION_MINOR, rialtomsesinks, "Sinks which communicate with RialtoServer",
-                  rialto_mse_sinks_init, "1.0", "LGPL", PACKAGE, "http://gstreamer.net/")
\ No newline at end of file
+GST_PLUGIN_DEFINE(GST_VERSION_MAJOR, GST_VERSION_MINOR, rialtosinks, "Sinks which communicate with RialtoServer",
+                  rialto_mse_sinks_init, "1.0", "LGPL", PACKAGE, "http://gstreamer.net/")
