Index: git/partner/dpi/gstreamer/ocdm/OCDMDrmSystem.h
===================================================================
--- git.orig/partner/dpi/gstreamer/ocdm/OCDMDrmSystem.h
+++ git/partner/dpi/gstreamer/ocdm/OCDMDrmSystem.h
@@ -12,6 +12,7 @@
 
 #include "playready_includes.h"
 
+#include <gst/gstcaps.h>
 #include <opencdm/open_cdm.h>
 #include <opencdm/open_cdm_ext.h>
 #include <opencdm/open_cdm_adapter.h>
