Source: Liberty Global Service B.V.
Index: git/rdk_gstreamer_utils.cpp
===================================================================
--- git.orig/rdk_gstreamer_utils.cpp
+++ git/rdk_gstreamer_utils.cpp
@@ -78,7 +78,7 @@ namespace rdk_gstreamer_utils {
     {
         const char* audiodecodername = getAudioDecoderName_soc();
         GstElement* audiodecoder = retrieveGstElementByName(pipeline, audiodecodername);
-        GstElement* videodecoder = retrieveGstElementByName(pipeline, "westerossink"); //default on RDK platforms
+        GstElement* videodecoder = retrieveGstElementByName(pipeline, "rialtomsevideosink");
         const char* AudioUnderflowSignal = getAudioUnderflowSignalName_soc();
         const char* VideoUnderflowSignal = getVideoUnderflowSignalName_soc();
 
Index: git/rialto/rdk_gstreamer_utils_soc.cpp
===================================================================
--- git.orig/rialto/rdk_gstreamer_utils_soc.cpp
+++ git/rialto/rdk_gstreamer_utils_soc.cpp
@@ -77,8 +77,8 @@ namespace rdk_gstreamer_utils
     GstElement * configureUIAudioSink_soc(bool TTSenabled)
     {
         GstElement *audioSink = NULL;
-        LOG_RGU("configureUIAudioSink_soc: : connecting rialtouiaudiosink");
-        audioSink = gst_element_factory_make ("rialtouiaudiosink","rialtouiaudiosink");
+        LOG_RGU("configureUIAudioSink_soc: : connecting rialtowebaudiosink");
+        audioSink = gst_element_factory_make ("rialtowebaudiosink","rialtowebaudiosink");
 
         return audioSink;
     }
@@ -190,7 +190,7 @@ namespace rdk_gstreamer_utils
 
     bool getDelayTimerEnabled_soc()
     {
-        return true;
+        return false;
     }
 
     void SetAudioServerParam_soc(bool enabled)
