Source: Liberty Global Service B.V.
Index: git/partner/dpi/gstreamer/FileSystem.h
===================================================================
--- git.orig/partner/dpi/gstreamer/FileSystem.h
+++ git/partner/dpi/gstreamer/FileSystem.h
@@ -28,25 +28,8 @@
 #include "Module.h"
 
 #include "ThunderLink.h"
-/* TODO: RDKservices event handling seems to have a bug whereby Api invocations fail after an event is received. Resorting to
-IARM based event handling till this is resolved. DELIA-43952 */
-#if defined(USE_DISPLAY_SETTINGS) || !defined (ENABLE_THUNDER_EVENTS)
-#include <libIARM.h>
-#include <libIBus.h>
-#include "libIBusDaemon.h"
-#include "dsMgr.h"
-#include "dsDisplay.h"
-#include <iarmUtil.h>
-#include "pwrMgr.h"
-
-static void dsHdmiEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len);
-static void DisplResolutionHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len);
-static void AudioModeHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len);
-static void PowerModeHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len);
-#elif defined(ENABLE_THUNDER_EVENTS)
 //Thunder Event handler
 static void thunderEventCallback(netflix::device::thunderlink::Event eventtype);
-#endif
 std::string base64_decode(const std::string &in);
 void cec_callback(bool status);
 
Index: git/partner/dpi/gstreamer/FileSystem.cpp
===================================================================
--- git.orig/partner/dpi/gstreamer/FileSystem.cpp
+++ git/partner/dpi/gstreamer/FileSystem.cpp
@@ -261,18 +261,9 @@ FileSystem::FileSystem(std::string *erro
     /*TODO: RDKservices event handling seems to have a bug whereby Api invocations fail after an event is received. Resorting to
      IARM based event handling till this is resolved. DELIA-43952 */
 
-#if defined(USE_DISPLAY_SETTINGS) || !defined(ENABLE_THUNDER_EVENTS)
-    printf("registering DSMGR events...\n");
-    IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG, dsHdmiEventHandler);
-    IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDCP_STATUS, dsHdmiEventHandler);
-    IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_RES_POSTCHANGE, DisplResolutionHandler);
-    IARM_Bus_RegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_AUDIO_MODE, AudioModeHandler);
-    IARM_Bus_RegisterEventHandler(IARM_BUS_PWRMGR_NAME,IARM_BUS_PWRMGR_EVENT_MODECHANGED, PowerModeHandler);
-#elif defined (ENABLE_THUNDER_EVENTS)
     printf("subscribing to thunder events...\n");
     syslog(LOG_INFO, "%s: %d ThunderLink : Event Notifications from Thunder : %s", __FUNCTION__, __LINE__, \
 	ThunderLink::getinstance()->registerCallback(thunderEventCallback) == WPEFramework::Core::ERROR_NONE ? "Enabled" : "Failed");
-#endif
 
      ThunderLink::getinstance()->cec_registerCallback(cec_callback);
     suspendOnStart_ = suspendModeOnStart();
@@ -284,45 +275,6 @@ FileSystem::FileSystem(std::string *erro
 
 }
 
-#if defined(USE_DISPLAY_SETTINGS) || !defined(ENABLE_THUNDER_EVENTS)
-static void DisplResolutionHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
-{
-
-  printf("%s enter\n", __FUNCTION__);
-  switch (eventId) {
-    case IARM_BUS_DSMGR_EVENT_RES_PRECHANGE:
-      printf("Received IARM_BUS_DSMGR_EVENT_RES_PRECHANGE \n");
-      break;
-    case IARM_BUS_DSMGR_EVENT_RES_POSTCHANGE:
-      {
-#ifdef SUPPORT_NATIVEFRAMERATE
-        if (sConfiguration->supportFollowVideoSourceFrameRate != ISystem::FOLLOW_VIDEO_SOURCE_FRAME_RATE_ENABLED)
-#endif
-        {
-          int dw = 1280;
-          int dh = 720;
-
-          IARM_Bus_DSMgr_EventData_t *eventData = (IARM_Bus_DSMgr_EventData_t *)data;
-          dw = eventData->data.resn.width ;
-          dh = eventData->data.resn.height ;
-
-          printf("Received IARM_BUS_DSMGR_EVENT_RES_POSTCHANGE  event width:%d height:%d\n", dw, dh);
-
-          if(nrdApp() && nrdApp()->getDeviceLib() && nrdApp()->getSystem())
-          {
-            FileSystem* fs = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-            if(fs) fs->setSupportedVideoOutputs(false);
-          }
-        }
-      }
-      break;
-
-    default:
-      printf("Received unknown event %d\n", eventId);
-      break;
-  }
-}
-
 void cec_callback(bool status)
 {
         syslog(LOG_WARNING,"%s %s: %d - Received Callback",  __FILE__, __FUNCTION__, __LINE__);
@@ -349,118 +301,6 @@ void FileSystem::UpdateCecStatus()
   }
 }
 
-void AudioModeHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
-{
-  printf("%s enter Event id is %d\n", __FUNCTION__, eventId);
-  switch (eventId) {
-    case IARM_BUS_DSMGR_EVENT_AUDIO_MODE:
-      {
-        printf("Received IARM_BUS_DSMGR_EVENT_AUDIO_MODE \n");
-
-        if(nrdApp() && nrdApp()->getDeviceLib() && nrdApp()->getSystem())
-        {
-          FileSystem* fs = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-          if(fs) fs->setSupportedAudioOutputs();
-        }
-      }
-      break;
-    default:
-      printf("Received unknown event %d\n", eventId);
-      break;
-  }
-}
-
-void PowerModeHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
-{
-  printf("%s enter Event id is %d\n", __FUNCTION__, eventId);
-  if (eventId == IARM_BUS_PWRMGR_EVENT_MODECHANGED ) {
-    IARM_Bus_PWRMgr_EventData_t *param = (IARM_Bus_PWRMgr_EventData_t *)data;
-    printf("Event IARM_BUS_PWRMGR_EVENT_MODECHANGED: State Changed %d -- > %d\n",
-                     param->data.state.curState, param->data.state.newState);
-    if (((param->data.state.curState == IARM_BUS_PWRMGR_POWERSTATE_STANDBY) || (param->data.state.curState == IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP))
-    && (param->data.state.newState == IARM_BUS_PWRMGR_POWERSTATE_ON)) {
-      standbyWakeup=true;
-      time(&gwakeuptime);
-    }
-    else if ((param->data.state.curState == IARM_BUS_PWRMGR_POWERSTATE_ON) && ((param->data.state.newState == IARM_BUS_PWRMGR_POWERSTATE_STANDBY) || (param->data.state.newState == IARM_BUS_PWRMGR_POWERSTATE_STANDBY_LIGHT_SLEEP)))
-    {
-      isActive = false;
-      if(nrdApp() && nrdApp()->getDeviceLib() && nrdApp()->getSystem())
-      {
-          FileSystem* fs = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-          if(fs) fs->UpdateCecStatus();
-      }
-    }
-
-  }
-}
-
-void dsHdmiEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
-{
-  printf("%s enter\n", __FUNCTION__);
-  switch (eventId) {
-    case IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG :
-      {
-
-        IARM_Bus_DSMgr_EventData_t *eventData = (IARM_Bus_DSMgr_EventData_t *)data;
-        int hdmi_hotplug_event = eventData->data.hdmi_hpd.event;
-
-        const char *hdmihotplug = (hdmi_hotplug_event == dsDISPLAY_EVENT_CONNECTED) ? "connected" : "disconnected";
-        printf("Received IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG  event data:%d status: %s\n", hdmi_hotplug_event, hdmihotplug);
-
-        if(hdmi_hotplug_event != dsDISPLAY_EVENT_CONNECTED) {
-          // Disconnected, set the HDMI Preferences to auto
-          printf("Received IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG Disconnected calling SetHdmiPreferences\n");
-	  //Thunder call to SetHdmiPreferences from SOCUtility
-          DeviceLibNative::getSOCUtilitiesInstance()->SetHdmiPreferences(HDCP_MAX);
-        }
-#ifdef SUPPORT_NATIVEFRAMERATE
-        else {
-          FileSystem* fileSystem = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-          if (fileSystem)
-          {
-            fileSystem->mSystemMonitor->clearDynamicFrameRateMap();
-            fileSystem->createDynamicFrameRateMap();
-          }
-        }
-#endif
-
-        if(nrdApp() && nrdApp()->getDeviceLib() && nrdApp()->getSystem())
-        {
-          FileSystem* fs = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-          if(fs) fs->setSupportedVideoOutputs(false);
-          if(fs) fs->setSupportedAudioOutputs();
-        }
-
-      }
-      break;
-
-    case IARM_BUS_DSMGR_EVENT_HDCP_STATUS :
-      {
-#ifdef SUPPORT_NATIVEFRAMERATE
-        if (sConfiguration->supportFollowVideoSourceFrameRate != ISystem::FOLLOW_VIDEO_SOURCE_FRAME_RATE_ENABLED)
-#endif
-        {
-          IARM_Bus_DSMgr_EventData_t *eventData = (IARM_Bus_DSMgr_EventData_t *)data;
-          int hdcpStatus = eventData->data.hdmi_hdcp.hdcpStatus;
-          const char *hdcpStatusStr = (hdcpStatus == dsHDCP_STATUS_AUTHENTICATED) ? "authenticated" : "authentication failure";
-          printf("Received IARM_BUS_DSMGR_EVENT_HDCP_STATUS  event data:%d status:%s\n", hdcpStatus, hdcpStatusStr);
-
-          if(nrdApp() && nrdApp()->getDeviceLib() && nrdApp()->getSystem())
-          {
-            FileSystem* fs = static_cast<FileSystem*>(nrdApp()->getSystem().get());
-            if(fs) fs->setSupportedVideoOutputs(false);
-         }
-        }
-      }
-      break;
-
-    default:
-      printf("Received unknown IARM bus event:%d\n", eventId);
-      break;
-  }
-}
-#elif defined(ENABLE_THUNDER_EVENTS) //enable Thunder event handlers when DELIA-43952 is resolved
 static void thunderEventCallback(thunderlink::Event event)
 {
   FileSystem* fs;
@@ -495,7 +335,6 @@ static void thunderEventCallback(thunder
       syslog(LOG_ERR, "%s: %d Received Unknown Thunder event\n", __FUNCTION__, __LINE__);
   }
 }
-#endif
 
 FileSystem::~FileSystem()
 {
@@ -743,7 +582,7 @@ static bool x_AddHDProfile(std::vector<C
   return retVal;
 }
 
-static bool x_ConfirmHDRSupport(std::vector<ContentProfile>& profiles, vector<HdrFormats> capabilities, bool bHDR10, bool bDolbyVision)
+static bool x_ConfirmHDRSupport(std::vector<ContentProfile>& profiles, std::vector<HdrFormats> capabilities, bool bHDR10, bool bDolbyVision)
 {
   bool retVal = false;
   if(bHDR10) {
@@ -897,7 +736,7 @@ void FileSystem::setSupportedVideoOutput
   uint32_t heightCM         = 0;
   uint32_t widthCM          = 0;
 
-  vector<HdrFormats> hdrCapabilities;
+  std::vector<HdrFormats> hdrCapabilities;
   PlaybackResolution _Resolution;
   HdcpParams params;
 
@@ -1047,127 +886,6 @@ else {
         syslog(LOG_INFO,"%s: %d : isActive:%d", __FUNCTION__, __LINE__, isActive);
       }
       videoOutputState.cecState = isActive ? CEC_ACTIVE : CEC_INACTIVE;
-#if defined(USE_DISPLAY_SETTINGS) || !defined(ENABLE_THUNDER_EVENTS)
-      try {
-            int dsFrameRate = 0;
-            int tValue = 0;
-            int tColorDepth = 0;
-            ::device::VideoOutputPort vPort = ::device::Host::getInstance().getVideoOutputPort("HDMI0");
-
-            dsFrameRate = vPort.getResolution().getFrameRate().getId();
-            switch (dsFrameRate)
-            {
-              case dsVIDEO_FRAMERATE_24:         /**< Played at 24 frames per second.           */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_24;
-                break;
-              case dsVIDEO_FRAMERATE_25:         /**< Played at 25 frames per second.           */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_25;
-                break;
-              case dsVIDEO_FRAMERATE_30:         /**< Played at 30 frames per second.           */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_30;
-                break;
-              case dsVIDEO_FRAMERATE_60:         /**< Played at 60 frames per second.           */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_60;
-                break;
-              case dsVIDEO_FRAMERATE_23dot98:    /**< Played at 23.98 frames per second.        */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_23_976;
-                break;
-              case dsVIDEO_FRAMERATE_29dot97:    /**< Played at 29.97 frames per second.        */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_29_97;
-                break;
-              case dsVIDEO_FRAMERATE_50:         /**< Played at 50 frames per second.           */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_50;
-                break;
-              case dsVIDEO_FRAMERATE_59dot94:    /**< Played at 59.94 frames per second.        */
-                videoOutputState.frameRate = FrameRate::FRAMERATE_59_94;
-                break;
-              case dsVIDEO_FRAMERATE_UNKNOWN:    /**< Unknown frame rate.                       */
-              default:
-                videoOutputState.frameRate = FrameRate::FRAMERATE_UNKNOWN;
-                break;
-            }
-
-            //Format
-            tValue = vPort.getColorSpace();
-            //Log::warn(TRACE_DPI, "[%s:%d] vPort.getColorSpace() retuned %d",__FILE__, __LINE__, tValue);
-            if (tValue == dsDISPLAY_COLORSPACE_YCbCr420)
-                videoOutputState.format = Format::FORMAT_YCBCR_420;
-            else if (tValue == dsDISPLAY_COLORSPACE_YCbCr422)
-                videoOutputState.format = Format::FORMAT_YCBCR_422;
-            else if (tValue == dsDISPLAY_COLORSPACE_YCbCr444)
-                videoOutputState.format = Format::FORMAT_YCBCR_444;
-            else if (tValue == dsDISPLAY_COLORSPACE_RGB)
-                videoOutputState.format = Format::FORMAT_RGB_444;
-            else if (tValue == dsDISPLAY_COLORSPACE_AUTO)
-                videoOutputState.format = Format::FORMAT_OTHER;
-            else
-                videoOutputState.format = Format::FORMAT_UNKNOWN;
-
-            tColorDepth = vPort.getColorDepth();
-            //Log::warn(TRACE_DPI, "[%s:%d] vPort.getColorDepth() retuned %d",__FILE__, __LINE__, tColorDepth);
-            if (videoOutputState.format == Format::FORMAT_YCBCR_422)
-            {
-                videoOutputState.colorDepth = ColorDepth::COLORDEPTH_12_BIT;
-                //Log::warn(TRACE_DPI, "[%s:%d] set videoOutputState.colorDepth = 12",__FILE__, __LINE__);
-            }
-            else
-            {
-                videoOutputState.colorDepth = static_cast<ColorDepth>(tColorDepth);
-            }
-
-
-            //matrixCoefficients
-            tValue = vPort.getMatrixCoefficients();
-            //Log::warn(TRACE_DPI, "[%s:%d] vPort.getMatrixCoefficients() retuned %d",__FILE__, __LINE__, tValue);
-
-            if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_SMPTE_170M)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_SMPTE170M;
-            else if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_BT_709)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_BT709;
-            else if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_eXvYCC_601)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_XVYCC601;
-            else if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_XvYCC_709)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_XVYCC709;
-            else if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_BT_2020_NCL)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_BT2020RGB_YCBCR;
-            else if (tValue == dsDISPLAY_MATRIXCOEFFICIENT_BT_2020_CL)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_BT2020YCCBCBRC;
-            else if (tValue != dsDISPLAY_MATRIXCOEFFICIENT_UNKNOWN)
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_OTHER;
-            else
-                videoOutputState.colorimetry = Colorimetry::COLORIMETRY_UNKNOWN;
-
-            // videoEOTF
-            tValue = vPort.getVideoEOTF();
-            //Log::warn(TRACE_DPI, "[%s:%d] vPort.getVideoEOTF() retuned %d",__FILE__, __LINE__, tValue);
-
-
-            if (tValue == dsHDRSTANDARD_NONE)
-                videoOutputState.eotf = Eotf::EOTF_BT1886;
-            else if ((tValue == dsHDRSTANDARD_HLG) || (tValue == (dsHDRSTANDARD_HLG|dsHDRSTANDARD_HDR10)))
-                videoOutputState.eotf = Eotf::EOTF_BT2100;
-            else if (tValue == dsHDRSTANDARD_HDR10)
-                videoOutputState.eotf = Eotf::EOTF_SMPTE_ST_2084;
-            else if (tValue == dsHDRSTANDARD_DolbyVision)
-                videoOutputState.eotf = Eotf::EOTF_OTHER;
-
-
-            //quantizationRange
-            tValue = vPort.getQuantizationRange();
-            //Log::warn(TRACE_DPI, "[%s:%d] vPort.getQuantizationRange() retuned %d",__FILE__, __LINE__, tValue);
-
-            if (tValue == dsDISPLAY_QUANTIZATIONRANGE_UNKNOWN)
-                videoOutputState.quantizationRange = QuantizationRange::QUANTIZATIONRANGE_UNKNOWN;
-            else if (tValue == dsDISPLAY_QUANTIZATIONRANGE_LIMITED)
-                videoOutputState.quantizationRange = QuantizationRange::QUANTIZATIONRANGE_LIMITED;
-            else if (tValue == dsDISPLAY_QUANTIZATIONRANGE_FULL)
-                videoOutputState.quantizationRange = QuantizationRange::QUANTIZATIONRANGE_FULL;
-
-            Log::warn(TRACE_DPI, "[%s:%d] vPort FrameRate %d, QuantizationRange %d, Colorimetry %d, EOTF %d, Format %d, ColorDepth %d(%d)",__FILE__, __LINE__, (int)videoOutputState.frameRate, (int)videoOutputState.quantizationRange, (int)videoOutputState.colorimetry, (int)videoOutputState.eotf, (int)videoOutputState.format, (int)videoOutputState.colorDepth, tColorDepth);
-      }catch (const std::exception e) {
-            syslog(LOG_NOTICE,"exception from devicesettings vPort fail....%s\n", e.what());
-      }
-#endif
       mActiveVideoOutputOverride.push_back(videoOutputState);
       mSystemMonitor->setActiveVideoOutput(mActiveVideoOutputOverride);
     }
@@ -1392,15 +1110,6 @@ void FileSystem::init(std::shared_ptr<IS
 
 void FileSystem::shutdown()
 {
-#if defined (USE_DISPLAY_SETTINGS) || !defined (ENABLE_THUNDER_EVENTS)
-  printf("filesystem()::shutdown() unregistering IARM DSMGR events...\n");
-  IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDMI_HOTPLUG);
-  IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_HDCP_STATUS);
-  IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_RES_POSTCHANGE);
-  IARM_Bus_UnRegisterEventHandler(IARM_BUS_DSMGR_NAME,IARM_BUS_DSMGR_EVENT_AUDIO_MODE);
-  IARM_Bus_UnRegisterEventHandler(IARM_BUS_PWRMGR_NAME,IARM_BUS_PWRMGR_EVENT_MODECHANGED);
-#endif
-
     mCommonSystem->shutdown();
     mEventDispatcher.reset();
 }
@@ -2381,61 +2090,6 @@ void FileSystem::createDynamicFrameRateM
 std::list<FrameRate> FileSystem::getFramerates()
 {
   std::list<FrameRate> frameRateList;
-#if defined(USE_DISPLAY_SETTINGS) || !defined(ENABLE_THUNDER_EVENTS)
-
-  std::map<FrameRate, FrameRate>::iterator it;
-
-  PlaybackResolution currentResolution = RESOLUTION_UNKNOWN;
-
-  uint32_t ret = ThunderLink::getinstance()->getCurrentResolutionFromDispSettings(currentResolution);
-  std::string currentResolutionStr = ThunderLink::getinstance()->getResolutionToString(currentResolution);
-
-  ::device::VideoOutputPort &vPort = ::device::Host::getInstance().getVideoOutputPort("HDMI0");
-  const ::device::List<::device::VideoResolution > supportedResolutions = ::device::VideoOutputPortConfig::getInstance().getPortType(vPort.getType().getId()).getSupportedResolutions();
-  for (size_t j = 0; j < supportedResolutions.size(); j++)
-  {
-     const ::device::VideoResolution &resolution = supportedResolutions.at(j);
-     std::string resolutionName = resolution.getName();
-
-     if (resolutionName.find(currentResolutionStr) != string::npos)
-     {
-       int dsFrameRate = resolution.getFrameRate().getId();
-       switch (dsFrameRate)
-       {
-          case dsVIDEO_FRAMERATE_23dot98:
-            frameRateList.push_back(FRAMERATE_23_976);
-            break;
-          case dsVIDEO_FRAMERATE_24:
-            frameRateList.push_back(FRAMERATE_24);
-            break;
-          case dsVIDEO_FRAMERATE_25:
-            frameRateList.push_back(FRAMERATE_25);
-            break;
-          case dsVIDEO_FRAMERATE_29dot97:
-            frameRateList.push_back(FRAMERATE_29_97);
-            break;
-          case dsVIDEO_FRAMERATE_30:
-            frameRateList.push_back(FRAMERATE_30);
-            break;
-          case dsVIDEO_FRAMERATE_50:
-            frameRateList.push_back(FRAMERATE_50);
-            break;
-          case dsVIDEO_FRAMERATE_59dot94:
-            frameRateList.push_back(FRAMERATE_59_94);
-            break;
-          case dsVIDEO_FRAMERATE_60:
-            frameRateList.push_back(FRAMERATE_60);
-            break;
-          case dsVIDEO_FRAMERATE_UNKNOWN:
-          default:
-            printf("%s::%s:%d  dsVIDEO_FRAMERATE_UNKNOWN\n",__FILE__, __FUNCTION__, __LINE__) ;
-            break;
-       }
-     }
-  }
-
-
-#endif
   return frameRateList;
 }
 
Index: git/partner/dpi/gstreamer/FilePreApp.cpp
===================================================================
--- git.orig/partner/dpi/gstreamer/FilePreApp.cpp
+++ git/partner/dpi/gstreamer/FilePreApp.cpp
@@ -22,7 +22,6 @@
 #include <stdint.h>
 #include <string.h>
 #include <errno.h>
-#include <rtRemote.h>
 #include <nrd/NrdApplication.h>     // For getEsn()
 #include <nrd/ISystem.h>
 #if defined(USE_VAP_PLUGINS)
Index: git/partner/dpi/gstreamer/SOCUtilities.cpp
===================================================================
--- git.orig/partner/dpi/gstreamer/SOCUtilities.cpp
+++ git/partner/dpi/gstreamer/SOCUtilities.cpp
@@ -9,7 +9,6 @@
 #include "gst_svp_meta.h"
 #endif
 
-using namespace device;
 using namespace netflix;
 using namespace netflix::device;
 using namespace netflix::device::esplayer;
Index: git/partner/dpi/gstreamer/CMakeLists.txt
===================================================================
--- git.orig/partner/dpi/gstreamer/CMakeLists.txt
+++ git/partner/dpi/gstreamer/CMakeLists.txt
@@ -153,7 +153,6 @@ if("${RDK_SVP}" STREQUAL "ENABLED")
 endif()
 
 include_directories($ENV{STAGING_DIR}/usr/include/WPEFramework)
-list(APPEND LIBRARIES IARMBus)
 list(APPEND LIBRARIES WPEFrameworkCore WPEFrameworkDefinitions WPEFrameworkProtocols WPEFrameworkPlugins WPEFrameworkSecurityUtil)
 list(APPEND SOURCES ThunderLink.cpp)
 list(APPEND SOURCES Module.cpp)
Index: git/partner/dpi/gstreamer/DeviceLibNative.cpp
===================================================================
--- git.orig/partner/dpi/gstreamer/DeviceLibNative.cpp
+++ git/partner/dpi/gstreamer/DeviceLibNative.cpp
@@ -649,7 +649,7 @@ static void x_SetAdditionalConfiguration
       pConfig->hevcEnabled     = true;
 
       // Does the device (not the TV) support HDR
-      vector<HdrFormats> hdr;
+      std::vector<HdrFormats> hdr;
       if(WPEFramework::Core::ERROR_NONE == ThunderLink::getinstance()->getSettopHDRSupport(hdr))
       {
         for(auto it = hdr.begin(); it != hdr.end(); it++)
Index: git/partner/dpi/gstreamer/ThunderLink.cpp
===================================================================
--- git.orig/partner/dpi/gstreamer/ThunderLink.cpp
+++ git/partner/dpi/gstreamer/ThunderLink.cpp
@@ -731,7 +731,7 @@ namespace thunderlink {
         //std::string externalNetworkFile = Configuration::dataWritePath() + "/network/external_network.json";
         std::string externalNetworkFile = "/opt/netflix/network/external_network.json";
 
-        int fd = open(externalNetworkFile.c_str(), O_CREAT | O_WRONLY | O_CLOEXEC);
+        int fd = open(externalNetworkFile.c_str(), O_CREAT | O_WRONLY | O_CLOEXEC, 0666);
         if(fd < 0)
         {
             syslog(LOG_WARNING, "[%s:%d] Cannot open %s; error %s",__FILENAME__, __LINE__, externalNetworkFile.c_str(), strerror(errno));
@@ -776,7 +776,7 @@ namespace thunderlink {
                 //std::string externalNetworkFile = Configuration::dataWritePath() + "/network/external_network.json";
                 std::string externalNetworkFile = "/opt/netflix/network/external_network.json";
 
-                int fd = open(externalNetworkFile.c_str(), O_CREAT | O_WRONLY | O_CLOEXEC);
+                int fd = open(externalNetworkFile.c_str(), O_CREAT | O_WRONLY | O_CLOEXEC, 0666);
                 if(fd < 0)
                 {
                     syslog(LOG_WARNING, "[%s:%d] Cannot open %s; error %s",__FILENAME__, __LINE__, externalNetworkFile.c_str(), strerror(errno));
